src/Libraries/DSOffice/Excel.cs

```diff
@@ -269,17 +269,28 @@ public static WorkBook SaveAsExcelWorkbook(WorkBook workbook, string filename)
         /// <param name="file">File representing the Microsoft Excel spreadsheet.</param>
         /// <param name="sheetName">Name of the worksheet containing the data.</param>
         /// <param name="readAsStrings">toggle to switch between reading Excel file as strings only or not</param>
+        /// <param name="visible">toggle to switch between showing and hiding the main Excel window</param>
         /// <returns name="data">Rows of data from the Excel worksheet.</returns>
         /// <search>office,excel,spreadsheet,ifequalreturnindex</search>
         [IsVisibleInDynamoLibrary(false)]
-        public static object[][] ReadFromFile(FileInfo file, string sheetName, bool readAsStrings = false)
+        public static object[][] ReadFromFile(FileInfo file, string sheetName, bool readAsStrings = false, bool visible = true)
         {
-            WorkBook wb = WorkBook.ReadExcelFile(file.FullName);
+        	object[][] data;
+        	
+        	if(!visible)
+        		ExcelInterop.ShowOnStartup = false;
```

Comment:
Our coding convention recommends putting statements in `if-else` blocks inside parentheses even if they are single statements, like:
```
if(!visible)
{
    ExcelInterop.ShowOnStartup = false;
}
```